src/Circle.cpp

```diff
@@ -12,6 +12,61 @@ Circle Circle::Unit()
 {
     return Circle(0, 0, 1);
 }
+
+std::vector<base::Vector2d> Circle::intersect(const Eigen::ParametrizedLine<double, 2>& line) const
+{
+    //Variable names correspond to names in:
+    //http://mathworld.wolfram.com/Circle-LineIntersection.html
+    
+    std::vector<base::Vector2d> result;
+    
+    Eigen::Vector2d p1 = line.pointAt(0);
```

Comment:
`const Eigen::Vector2d p1 = line.origin() - center;`
And actually, no need for `p2`